src/hocon_schema.erl

```diff
@@ -350,3 +357,36 @@ assert_unique_field_names([{Name, _} | Rest], Acc) ->
         false ->
             assert_unique_field_names(Rest, [BinName | Acc])
     end.
+
+new_cache(undefined) -> #{file => undefined, tab => undefined};
+new_cache(File) ->
+  case hocon:load(File) of
+    {ok, Schema} ->
+      Tab = ets:new(?MODULE, [set]),
+      ets:insert(Tab, {File, Schema}),
+      #{tab => Tab, file => File};
+    {error, Reason} ->
+      error(#{reason => Reason, file => File})
+  end.
+
+delete_cache(#{tab := undefined}) -> ok;
+delete_cache(#{tab := Tab}) -> ets:delete(Tab).
+
+resolve_schema(?DESC(Mod, Id), State) ->
```

Comment:
resolve_desc is a better name?

Reply:
we already have `field_schema(Schema,Key)`, 
so we can do `field_schema(Schema,default), field_schema(Schema, required) etc`
we don't choose `get_required_field(Schema)`, this make too many functions.

so I don't think resolve_desc is a better name, It may be resolve_xxx in the future.
Maybe `resolve_field`?

Reply:
Got it. resolve_schema makes sense.